src/main/java/com/spotifytelegramdeliverymessage/service/impl/BotServiceImpl.java

```diff
@@ -0,0 +1,92 @@
+package com.spotifytelegramdeliverymessage.service.impl;
+
+
+import com.spotifytelegramdeliverymessage.constant.BotCommands;
+import com.spotifytelegramdeliverymessage.constant.BotText;
+import com.spotifytelegramdeliverymessage.constant.SubscribeUserStatus;
+import com.spotifytelegramdeliverymessage.model.User;
+import com.spotifytelegramdeliverymessage.service.BotService;
+import com.spotifytelegramdeliverymessage.service.EmailService;
+import com.spotifytelegramdeliverymessage.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+import org.telegram.telegrambots.meta.api.methods.send.SendMessage;
+import org.telegram.telegrambots.meta.bots.AbsSender;
+import org.telegram.telegrambots.meta.exceptions.TelegramApiException;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@Service
+public class BotServiceImpl implements BotService {
+
+    @Lazy
+    @Autowired
+    private AbsSender absSender;
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private EmailService emailService;
+
+    private final Map<String, String> emailByChatId = new HashMap<>();
```

Comment:
what if your app is restarted? Consider adding this info also to db

Reply:
Done